### PR TITLE
Replace hashmap cache to LRU cache

### LINKED ISSUE #1576: High memory usage while syncing
During the block synchronization, codechain uses main memory over 8GB.
And the occupied memory was not cleared after the synchronization finished.

### PR DESCRIPTION
**Pull request by HoOngEe**
it closes #1576

### CODE REVIEW

File: core/src/blockchain/headerchain.rs [hunk 1]
```diff
@@ -99,7 +100,7 @@ impl HeaderChain {
             best_header_hash: RwLock::new(best_header_hash),
             best_proposal_header_hash: RwLock::new(best_proposal_header_hash),
 
-            header_cache: RwLock::new(HashMap::new()),
+            header_cache: Mutex::new(LruCache::new(1000)),
```
Comment (foriequal0) [inline core/src/blockchain/headerchain.rs]: Ditto

File: core/src/blockchain/body_db.rs [hunk 2]
```diff
@@ -48,7 +49,7 @@ impl BodyDB {
     /// Create new instance of blockchain from given Genesis.
     pub fn new(genesis: &BlockView, db: Arc<KeyValueDB>) -> Self {
         let bdb = Self {
-            body_cache: RwLock::new(HashMap::new()),
+            body_cache: Mutex::new(LruCache::new(1000)),
```
Comment (foriequal0) [inline core/src/blockchain/body_db.rs]: Would you separate the constant out?

File: core/src/blockchain/headerchain.rs (hunk 1 above)
Comment (HoOngEe) [reply core/src/blockchain/headerchain.rs]: Done

File: core/src/blockchain/body_db.rs (hunk 2 above)
Comment (HoOngEe) [reply core/src/blockchain/body_db.rs]: Ok, I applied

**Pull request merged**